tests/reg_tests/reg_test_classes.py

```diff
@@ -32,4 +43,14 @@ def train(self):
             test_func = getattr(self, test)
             test_func()
 
+            # Remove some path-related options before storing them as metadata
+            opts = copy.deepcopy(self.CFDSolver.getOptions())
```

Comment:
Two questions:
1) `getOptions()` already returns a copy of the dictionary. Did you need `deepcopy` to get it to work?
2) Does `KeyError` ever get raised?

Reply:
To add to this, it's preferable to just do `opts.pop(key, None)` as a one-liner, which will not raise `KeyError` if `key` is not found.

Reply:
Also, I would prefer to still keep in the paths, but convert them to relative paths (to the current working directory) so the personal/sensitive info is removed. I will give this a shot and push up soon.

Reply:
Sure, I have no strong opinions about this, probably retaining the relative path makes more sense - additional and potential useful info. About your previous point, I wanted to delete the key altogether, doesn't that command keep the key with value `None`?

@sseraj: 1) I thought the same but I had some "missing key" errors while training.. apparently I was not popping out the copy? Pretty sure I was already using `getOption` but we can double check; 2) I guess it's extra safety... I realize just now that we are printing all the options, when I first manually deleted my personal folder path only some had `restartFile not None` so I got tricked. Anyway, if we opt to keep the relative path, this is not going to be an issue anymore.

Reply:
Nope, `pop` actually is designed to return the value, and `None` will just specify that as the return value if `key` is not found, instead of raising the error. I also removed the extra copy just now -- I agree with @sseraj I don't think it's necessary.